### PR TITLE
feat(cxl-ui): add login styles

### PR DESCRIPTION
**Pull request by saas786**
https://app.clickup.com/t/3ccnz6j

### CODE REVIEW

**Comment by kylebrodeur** [comment]
Task linked: [CU-3ccnz6j Change lost my password link style](https://app.clickup.com/t/3ccnz6j)

File: packages/cxl-ui/scss/cxl-login.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.login #nav a,
+.login #backtoblog a {
+  color: #2271b1;
```
Comment (lkraav) [inline packages/cxl-ui/scss/cxl-login.scss:3]: Can you check what would be the size of a bundle that includes only cxl-lumo-styles/color.js?

Ideally we would be able to build a minimal bundle of at least color and typography for specialty screens like login.

**Review by lkraav — CHANGES_REQUESTED** [review]
* @pawelkmpt I would not accept any PRs without an accompanying story. Otherwise, validating whether any such shipped CSS makes any sense becomes super complicated, because you need make roundtrips to pulling Aybolit changes in wp-theme-lib.

* I'm not clear on why it's necessary to add new cxl-ui devDeps.

* Also not sure whether this new CSS build strategy makes sense as-is, or if it could be better :thinking: it would help if there was another CSS implementation example to use this system - what is an important non-web-component thing we need to style - Woo Checkout comes to mind immediately.

**Review by pawelkmpt — COMMENTED** [review]
I commented in CU but good to have it here as well.

> tell me what's the simplest and cleanest solution,

Is writing a SASS compile directives really the simplest and cleanest solution? Do we really need such for 10 lines of CSS code?

I doubt it.

File: packages/cxl-ui/scss/cxl-login.scss (hunk 1 above)
Comment (pawelkmpt) [reply packages/cxl-ui/scss/cxl-login.scss:3]: For now we don't need such advanced checks. We just need **red links** on login page. That's it. When redesign time come we'll think about system for it.

Comment (lkraav) [reply packages/cxl-ui/scss/cxl-login.scss:3]: > checks

Wdym "checks" here?

**Comment by lkraav** [comment]
> Is writing a SASS compile directives

What are "SASS compile directives"?

File: packages/cxl-ui/scss/cxl-login.scss (hunk 1 above)
Comment (pawelkmpt) [reply packages/cxl-ui/scss/cxl-login.scss:3]: I don't know honestly. I was tired.

**Comment by pawelkmpt** [comment]
> What are "SASS compile directives"?

```js
mix.sass(`${scssPath}/cxl-login.scss`, `${distPath}/styles/cxl-login.css`);
```

**Comment by saas786** [comment]
Replying to both `institute-theme` & `aybolit` PRs.

Originally created: https://github.com/conversionxl/aybolit/commit/110f6bd#diff-31fbf03c262224ebb188944a78ac8e2ac946dbbd6dfa52ef249d3772b74142f4

It pulls lumo colors & we can use our cxl brand color for links. But saw existing blue links, so reverted it.

chrome_n0KOvymsHy

> put all in Custom Login plugin admin screen [cxl.com/institute/wp-admin/options-general.php?page=custom-login](https://cxl.com/institute/wp-admin/options-general.php?page=custom-login) (all in one place for future refactor)

Using custom css inline is not ideal, as we can't utilize the lumo styles now or later.

> (preferred) or, get it all out of Custom Login, and put into a wp-theme-lib SCSS file, then disable plugin (major advantage: WPS will also gain a reference login experience)

Hmmmm, should create a new Login component in `wp-theme-lib` like `CXLUi/WPLogin.php`?

ref: https://github.com/conversionxl/wp-theme-lib/tree/master/src/CXLUi

Sure, can move custom css from `options-general.php?page=custom-login` to current PR.

> Can you check what would be the size of a bundle that includes only cxl-lumo-styles/color.js?
Ideally we would be able to build a minimal bundle of at least color and typography for specialty screens like login.

I have given it a thought, and I don't see us using any JS on WP Login page, so using a bundle is overkill, as my estimate was it will be at least 20KB+, 
where if we just serve css without cxl-ui bundle, its hardly gonna be 1KB+ or similar.
Even if we use: https://github.com/conversionxl/aybolit/commit/110f6bd#diff-31fbf03c262224ebb188944a78ac8e2ac946dbbd6dfa52ef249d3772b74142f4 (lumo colors).

> @pawelkmpt I would not accept any PRs without an accompanying story. Otherwise, validating whether any such shipped CSS makes any sense becomes super complicated, because you need make roundtrips to pulling Aybolit changes in wp-theme-lib.

Story only make sense, if you want to see 
https://www.figma.com/file/T7CzLFP2K5yHPShh1d7tlK/Login-Flow-Redesign?node-id=0%3A1&t=RYBfeGkrBn69I2HD-1 
in action, and it would require some time & effort to create story for each login screen view, it's an overkill for our custom css.

As far as I remember (can look for such threads), we wanted to move away from `wp-theme-lib` scss, and instead server styles via `aybolit`, GravityForm css was major step towards this?

> I'm not clear on why it's necessary to add new cxl-ui devDeps.
> Also not sure whether this new CSS build strategy makes sense as-is, or if it could be better 🤔 it would help if there was another CSS implementation example to use this system - what is an important non-web-component thing we need to style - Woo Checkout comes to mind immediately.

There is no new css build strategy, it was always there, we just had a need to server simple css via aybolit, now we have, so utilizing it. There is component specific css & there is none component css, and both have its own use and purpose.

---

Let me know your what you think.

**Comment by pawelkmpt** [comment]
> Using custom css inline is not ideal, as we can't utilize the lumo styles now or later (on login page).

I agree. But current goal is to improve login experience with low cost before Christmas. We're not looking for "ideal" now. We're looking for "better".

**Comment by lkraav** [comment]
> I have given it a thought, and I don't see us using any JS on WP Login page, so using a bundle is overkill, as my estimate was it will be at least 20KB+, where if we just serve css without cxl-ui bundle, its hardly gonna be 1KB+ or similar.
> Even if we use: https://github.com/conversionxl/aybolit/commit/110f6bd#diff-31fbf03c262224ebb188944a78ac8e2ac946dbbd6dfa52ef249d3772b74142f4 (lumo colors).

Noticed upstream https://github.com/vaadin/web-components/pull/4935

Even though login is a stupidly "special" screen, maybe one of its kind in the whole app - maybe spending that 20kB on the micro-bundle makes more sense than trying to build CSS bundles..?

> Story only make sense, if you want to see
https://www.figma.com/file/T7CzLFP2K5yHPShh1d7tlK/Login-Flow-Redesign?node-id=0%3A1&t=RYBfeGkrBn69I2HD-1
in action, and it would require some time & effort to create story for each login screen view, it's an overkill for our custom login css.

Story makes perfect sense if it only has the main login box, with no additional steps storied.

It's already much ahead of "nothing" by that point.

**Comment by pawelkmpt** [comment]
We don't need code solution now. Closing.

**Comment by saas786** [comment]
> Even though login is a stupidly "special" screen, maybe one of its kind in the whole app - maybe spending that 20kB on the micro-bundle makes more sense than trying to build CSS bundles..?

@lkraav 

It resulted in about `240kb+`.

explorer_bQXNAMpS3j

**Comment by lkraav** [comment]
> It resulted in about `240kb+`.

Quite disappointing, indeed. Even though I'm not sure what gets pulled into `vendor-login.js`, extracting CSS for cross-usage seems like the only viable option.

**Comment by saas786** [comment]
> > It resulted in about `240kb+`.
> 
> Quite disappointing, indeed. Even though I'm not sure what gets pulled into `vendor-login.js`, extracting CSS for cross-usage seems like the only viable option.

Just importing `colors,typography` packages now, and still seem to work, and around `32kb+` size.

explorer_YmZSqCKvW4

**Comment by lkraav** [comment]
> Just importing `colors,typography` packages now, and still seem to work, and around `32kb+` size.

That was the original idea. What did you import into the 240 kB package?

For 32 kB total, what does bundle analyzer show getting pulled in?

**Comment by saas786** [comment]
> That was the original idea. What did you import into the 240 kB package?

I wasn't sure if just importing `colors,typography` will be enough, 
just went with full `cxl-lumo-styles` import, which additionally had `themes, icons, global styles etc`.

> For 32 kB total, what does bundle analyzer show getting pulled in?

firefox_YiopGAXG5A

**Comment by lkraav** [comment]
> For 32 kB total, what does bundle analyzer show getting pulled in?

Yeah, `vaadin-themable-mixin` I also reached tracing by hand, and gave up a that point.

Unless we start needing other Lit-things, some CSS bundle build process is certainly more optimal.

Even though 80/20 + speed question arises: should we still load this small bundle, until we get the real thing done?

CSS bundle doesn't sound like 1-2 hr job :worried:

**Comment by saas786** [comment]
> CSS bundle doesn't sound like 1-2 hr job 😟

Its your preference,

we can use current JS bundle.

Or I can revert to custom css, which only includes css on login page.

**Pull request closed without merging**